libp2p/network/url_conversion.go

```diff
@@ -127,7 +127,30 @@ func getPeerIP(hostname string) (net.IP, error) {
 		if peerIP == nil {
 			continue
 		}
+		// TECHDEBT: remove this log line once direction is clearer
+		// on supporting multiple network addresses per peer.
+		if len(addrs) > 1 {
+			logger.Global.Warn().
+				Array("resolved", newStringLogArrayMarshaler(addrs)).
+				IPAddr("using", peerIP)
+		}
 		return peerIP, nil
 	}
 	return nil, newResolvePeerIPErr(hostname, err)
 }
+
+type stringLogArrayMarshaler struct {
```

Comment:
Since we created this struct for a super small log line, please define it inline.